src/rpc/ngHandlers/Tx.cpp

```diff
@@ -0,0 +1,121 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2023, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <rpc/RPCHelpers.h>
+#include <rpc/ngHandlers/Tx.h>
+
+namespace RPCng {
+
+TxHandler::Result
+TxHandler::process(Input input, boost::asio::yield_context& yield) const
+{
+    auto const rangeSupplied = input.minLedger && input.maxLedger;
+
+    if (rangeSupplied)
+    {
+        if (*input.minLedger > *input.maxLedger)
+            return Error{RPC::Status{RPC::RippledError::rpcINVALID_LGR_RANGE}};
+        if (*input.maxLedger - *input.minLedger > 1000)
+            return Error{
+                RPC::Status{RPC::RippledError::rpcEXCESSIVE_LGR_RANGE}};
+    }
```

Comment:
minor: This part could use a bit of spacing, maybe one line of space here would make blocks more stand out.